src/content-layout/index.tsx

```diff
@@ -0,0 +1,81 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useContext, useLayoutEffect, useRef } from 'react';
+import clsx from 'clsx';
+import { AppLayoutContext } from '../app-layout/visual-refresh/context';
+import { applyDisplayName } from '../internal/utils/apply-display-name';
+import { ContentLayoutProps } from './interfaces';
+import { getBaseProps } from '../internal/base-component';
+import useBaseComponent from '../internal/hooks/use-base-component';
+import { useContainerQuery } from '../internal/hooks/container-queries';
+import { useMergeRefs } from '../internal/hooks/use-merge-refs';
+import { useVisualRefresh } from '../internal/hooks/use-visual-mode';
+import styles from './styles.css.js';
+
+export { ContentLayoutProps };
+
+export default function ContentLayout({ children, disableOverlap, header, ...rest }: ContentLayoutProps) {
+  const baseProps = getBaseProps(rest);
+  const { breadcrumbs, setDynamicOverlapHeight } = useContext(AppLayoutContext);
+  const isVisualRefresh = useVisualRefresh();
+
+  const rootElement = useRef<HTMLDivElement>(null);
+  const { __internalRootRef } = useBaseComponent('ContentLayout');
+  const mergedRef = useMergeRefs(rootElement, __internalRootRef);
+
+  /**
+   * Observe the height of the ContentLayout background element. This value will be used to
+   * set the dynamicOverlapHeight property in the AppLayout which will extend the overlap
+   * height of the AppLayout down to include this element. This creates the appearance
+   * of a seamless dark background when these two components are used together.
+   */
+  const [overlapContainerQuery, overlapElement] = useContainerQuery(rect => rect.height);
+
+  useLayoutEffect(
+    function handleDynamicOverlapHeight() {
+      if (isVisualRefresh) {
+        setDynamicOverlapHeight(overlapContainerQuery ?? 0);
+      }
+    },
+    [isVisualRefresh, overlapContainerQuery, setDynamicOverlapHeight]
+  );
```

Comment:
Here we also need to reset the value before unmounting. We can use the new `useDynamicOverlap` hook introduced in https://github.com/cloudscape-design/components/pull/84

Reply:
Agreed I was just waiting on that PR to merge.